core/templates/domain/opportunity/translatable-texts.model.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Frontend Model for translatable texts
+ */
+
+export interface ContentIdToContentMapping {
+  [contentId: string]: string
+}
+
+export interface StateNamesToContentIdMapping {
+  [state: string]: ContentIdToContentMapping
+}
+
+export interface TranslatableTextsBackendDict {
+  'state_names_to_content_id_mapping': StateNamesToContentIdMapping;
+  'version': string;
+}
+
+export class TranslatableTexts {
+  constructor(
+      readonly stateNamesToContentIdMapping: StateNamesToContentIdMapping,
+      readonly version: string) {}
```

Comment:
Can we mark these as `private` as well?

Reply:
Done

Reply:
No, both are directly accessed in `translate-text.service.ts`.

Reply:
Ah, I see. Can you update those usages to use this class's getters instead? Then we should be able to mark these as private.